### DISCUSSION TITLE
Unable to compile PETSc with Mumps

### DISCUSSION THREAD

**Discussion by r-williams-2** [idaholab/moose#16441] [Q&A Getting Started] [answered]
Reposting [this](https://groups.google.com/g/moose-users/c/nQS1MdZrEgo/m/aV-a7Y3TAwAJ).

I was recently updating PETSc using the _update_and_rebuild_petsc.sh_ script. However, this failed at Mumps with the below error:

![image](https://user-images.githubusercontent.com/41152732/101692590-54e16700-3a68-11eb-8a6f-cc1be2828661.png)

In addition, please find attached the [configure.log](https://github.com/idaholab/moose/files/5669024/configure.log) file. It appears that the directory path for Mumps has not been created although I am not sure the reason for this. It should be noted that if I exclude mumps, PETSc compiles fine.

Thanks for your help.

Kind regards,
Ross.

**Comment by aeslaughter** [comment]
@fdkong

**Comment by fdkong** [comment, marked as answer]
New complied MUMPS conflict with Conda carried MUMPS. There are two options for you:

1) use conda to get all latest packages including PETSc and libMesh via  ` conda update --all`

or 

2) create a new environment without PETSc and libMesh in it. For example, `conda create --name newmoose moose-mpich moose-tools`. In this new environment, you can then use `update_and_rebuild_petsc.sh `. After you build PETSc, and you need to build libMesh. 

For conda users, we suggest taking option 1.

**Reply by r-williams-2** [reply]
So anaconda automatically manages PETSc as well? I wasn't aware of that. Thanks for clarifying that.

**Reply by fdkong** [reply]
Yes, `conda` manages PETSc and libMesh and other dependent packages.
